lrlex/src/lib/parser.rs

```diff
@@ -361,9 +365,73 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
         &self,
         off: usize,
         re_str: &'a str,
-    ) -> LexInternalBuildResult<(Vec<usize>, &'a str)> {
+    ) -> LexInternalBuildResult<(Vec<usize>, std::borrow::Cow<'a, str>)> {
         if !re_str.starts_with('<') {
-            Ok((vec![], re_str))
+            /// This implements the 'Table: Escape Sequences in lex' from POSIX lex specification
+            ///
+            /// Most of the escape handling is left to regex, except this part:
+            ///
+            /// Escape: \c
+            ///
+            /// Description:
+            /// A <backslash> character followed by any character not described in this table or in the table in
+            /// XBD File Format Notation ( '\\', '\a', '\b', '\f' , '\n', '\r', '\t', '\v' ).
+            ///
+            /// Meaning: The character 'c', unchanged.
+            fn unescape(re: Cow<str>) -> Cow<str> {
+                let re_str: &str = re.borrow();
+                let mut re_chars = re_str.char_indices().peekable();
+                debug_assert!(regex_syntax::is_meta_character('<').not());
+                let mut cursor = loop {
```

Comment:
I wouldn't mind a comment or two in here to explain to an idiot like me roughly what's going on :)

Reply:
Yeah, I was actually just in the process of writing one.

Reply:
So, perhaps I went overboard with the comments in [cf5b0b1](https://github.com/softdevteam/grmtools/pull/333/commits/cf5b0b177f76defff3a0afd47295226af0fc444e), but it is kind of a funny loop.